packages/yarnpkg-pnpify/sources/generateSdk.ts

```diff
@@ -147,168 +262,91 @@ class Wrapper {
-
-  await addVSCodeWorkspaceSettings(pnpApi, {
-    [`eslint.nodePath`]: npath.fromPortablePath(
-      ppath.dirname(ppath.dirname(ppath.dirname(
-        wrapper.getProjectPathTo(
-          `lib/api.js` as PortablePath,
-        ),
-      ))),
-    ),
-  });
-};
-
-export const generatePrettierWrapper = async (pnpApi: PnpApi, target: PortablePath) => {
-  const wrapper = new Wrapper(`prettier` as PortablePath, {pnpApi, target});
-
-  await wrapper.writeManifest();
-
-  await wrapper.writeBinary(`index.js` as PortablePath);
-
-  await addVSCodeWorkspaceSettings(pnpApi, {
-    [`prettier.prettierPath`]: npath.fromPortablePath(
-      wrapper.getProjectPathTo(
-        `index.js` as PortablePath,
-      ),
-    ),
-  });
-};
-
-export const generateTypescriptLanguageServerWrapper = async (pnpApi: PnpApi, target: PortablePath) => {
-  const wrapper = new Wrapper(`typescript-language-server` as PortablePath, {pnpApi, target});
+export const generateSdk = async (pnpApi: PnpApi, editors: typeof SUPPORTED_EDITORS | null, {report, base}: {report: Report, base: boolean}): Promise<void> => {
+  const topLevelInformation = pnpApi.getPackageInformation(pnpApi.topLevel)!;
+  const projectRoot = npath.toPortablePath(topLevelInformation.packageLocation);
 
-  await wrapper.writeManifest();
+  const targetFolder = ppath.join(projectRoot, PNPIFY_FOLDER);
+  const editorsFilePath = ppath.join(targetFolder, EditorsFile.fileName);
 
-  await wrapper.writeBinary(`lib/cli.js` as PortablePath);
-};
+  const hasEditorsFile = xfs.existsSync(editorsFilePath);
 
-const generateTypescriptWrapper = async (pnpApi: PnpApi, target: PortablePath) => {
-  const tsServerMonkeyPatch = `
-    tsserver => {
-      // VSCode sends the zip paths to TS using the "zip://" prefix, that TS
-      // doesn't understand. This layer makes sure to remove the protocol
-      // before forwarding it to TS, and to add it back on all returned paths.
-
-      const {isAbsolute} = require(\`path\`);
-
-      const Session = tsserver.server.Session;
-      const {onMessage: originalOnMessage, send: originalSend} = Session.prototype;
-
-      return Object.assign(Session.prototype, {
-        onMessage(/** @type {string} */ message) {
-          return originalOnMessage.call(this, JSON.stringify(JSON.parse(message), (key, value) => {
-            return typeof value === 'string' ? removeZipPrefix(value) : value;
-          }));
-        },
-
-        send(/** @type {any} */ msg) {
-          return originalSend.call(this, JSON.parse(JSON.stringify(msg, (key, value) => {
-            return typeof value === 'string' ? addZipPrefix(value) : value;
-          })));
-        }
-      });
-
-      function addZipPrefix(str) {
-        if (isAbsolute(str) && str.match(/\\.zip\\//) && !str.match(/^zip:/)) {
-          return \`zip:\${str}\`;
-        } else {
-          return str;
-        }
-      }
+  const userEditors = editors ?? new Set<SupportedEditor>();
```

Comment:
Rather than `userEditors`, I think something like `newEditors` or `requestedEditors` would be easier to understand.